packages/discovery-react-components/src/components/HtmlViewer/HtmlViewer.tsx

```diff
@@ -0,0 +1,262 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import React, {
+  FC,
+  useEffect,
+  useReducer,
+  useState,
+  Reducer,
+  SetStateAction,
+  Dispatch
+} from 'react';
+import { settings } from 'carbon-components';
+import { QueryResult } from 'ibm-watson/discovery/v2';
+import HtmlViewerDisplay from './components/HtmlViewerDisplay/HtmlViewerDisplay';
+import NavigationToolbar from './components/NavigationToolbar/NavigationToolbar';
+import { processDoc } from './utils/processDoc';
+import { getId, findElement } from 'utils/document/idUtils';
+import { withErrorBoundary, WithErrorBoundaryProps } from 'utils/hoc/withErrorBoundary';
+import { Field } from 'components/CIDocument/types';
+import { defaultTheme, Theme } from 'utils/theme';
+import { DocumentData, Section, EnrichmentField } from './types';
+import {
+  defaultMessages as HtmlViewerDefaultMsgs,
+  Messages as HtmlViewerMessages
+} from './messages';
+
+export type Messages = HtmlViewerMessages;
+
+const defaultMessages = {
+  ...HtmlViewerDefaultMsgs
+};
+
+export type ItemChangeFn = (item: string[]) => void;
+const noop = (): void => {};
+
+interface State {
+  isError: boolean;
+  styles: string[];
+  sections: Section[];
+  itemMap: {
+    byItem: any;
+    bySection: any;
+  };
+}
+
+enum ActionType {
+  RESET = 'RESET',
+  SET = 'SET',
+  ERROR = 'ERROR'
+}
+
+interface Action {
+  type: ActionType;
+  data: any;
+}
+
+const INITIAL_STATE: State = {
+  isError: false,
+  styles: [],
+  sections: [],
+  itemMap: { byItem: {}, bySection: {} }
+};
+
+const base = `${settings.prefix}--html-viewer`;
+
+export interface HtmlViewerProps extends WithErrorBoundaryProps {
+  /**
+   * Document data, as that returned by a query. Overrides result from SearchContext
+   */
+  document: QueryResult;
+  /**
+   * Specify which field in the document holds the text data to display
+   */
+  dataField?: string;
+  /**
+   * List of ids of elements to highight
+   */
+  // TODO: colbyj restrict this to string array (ids)?
+  highlightedList: any[];
+  /**
+   * Callback function on document processing success
```

Comment:
I believe this needs to be switched with the description below.

Reply:
Oops. 🤦‍♂